geode-core/src/main/java/org/apache/geode/internal/cache/map/RegionMapPut.java

```diff
@@ -396,12 +410,35 @@ private boolean checkUninitializedRegionPreconditions() {
   private boolean checkCreatePreconditions() {
     if (isIfNew()) {
       if (!getRegionEntry().isDestroyedOrRemoved()) {
+        // retain the version stamp of the existing entry for use in processing failures
+        EntryEventImpl event = getEvent();
+        if (event.getRegion().getConcurrencyChecksEnabled() &&
+            event.getOperation() == Operation.PUT_IF_ABSENT &&
+            !event.hasValidVersionTag() &&
+            event.isPossibleDuplicate()) {
+          if (ValueComparisonHelper.checkEquals(getRegionEntry().getValue(),
```

Comment:
instead of RegionEntry.getValue you could use getValueRetain. This is more performant for offheap regions. checkEquals is able to check the serialized value stored in off-heap memory (returned from getValueRetain as a StoredObject) whereas getValue copies the entire thing from offheap to the heap. If you use getValueRetain then you need to also add a finally block that calls OffHeapHelper.release(value)